### PR TITLE
chore: Generate nginx config using chatwoot installation scripts

### PR DESCRIPTION
**Pull request by sojan-official**
# Pull Request Template

## Description

Generate Nginx config using chatwoot installation scripts

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: deployment/chatwoot_nginx.conf [hunk 1]
```diff
@@ -0,0 +1,50 @@
+server {
+        listen 80;
```
Comment (pranavrajs) [inline deployment/chatwoot_nginx.conf:2]: Please use 2 space indentation.

File: deployment/chatwoot_nginx.conf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        listen [::]:443 ssl http2;
+        server_name chatwoot.domain.com www.chatwoot.domain.com;
+
+        underscores_in_headers on;
+
+        access_log /var/log/nginx/chatwoot_access_443.log;
+        error_log /var/log/nginx/chatwoot_error_443.log;
+
+        location / {
+            proxy_pass_header Authorization;
```
Comment (pranavrajs) [inline deployment/chatwoot_nginx.conf:23]: Please fix indentation.

File: deployment/chatwoot_nginx.conf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    		client_max_body_size 0;
+    		proxy_read_timeout 36000s;
+    		proxy_redirect off;
+        }
+
+        ssl_certificate /etc/letsencrypt/live/chatwoot.domain.com/fullchain.pem; # managed by Certbot
+        ssl_certificate_key /etc/letsencrypt/live/chatwoot.domain.com/privkey.pem; # managed by Certbot
+        add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
+
+    # curl https://ssl-config.mozilla.org/ffdhe4096.txt >> /etc/ssl/dhparam
```
Comment (pranavrajs) [inline deployment/chatwoot_nginx.conf:44]: Do we need this?

Comment (iriseden) [reply deployment/chatwoot_nginx.conf]: More secure

File: deployment/chatwoot_nginx.conf (hunk 1 above)
Comment (iriseden) [reply deployment/chatwoot_nginx.conf]: Done

File: deployment/chatwoot_nginx.conf (hunk 2 above)
Comment (iriseden) [reply deployment/chatwoot_nginx.conf]: Done

File: deployment/setup_20.04.sh [hunk 4]
```diff
@@ -76,8 +83,32 @@ cp /home/chatwoot/chatwoot/deployment/chatwoot.target /etc/systemd/system/chatwo
 systemctl enable chatwoot.target
 systemctl start chatwoot.target
 
-echo "Woot! Woot!! Chatwoot server installation is complete"
-echo "The server will be accessible at http://<server-ip>:3000"
-echo "To configure a domain and SSL certificate, follow the guide at https://www.chatwoot.com/docs/deployment/deploy-chatwoot-in-linux-vm"
-
 # TODO: Auto-configure Nginx with SSL certificate
+
+read -p 'Would you like to configure Webserver and SSL (yes or no): ' test
```
Comment (vishnu-narayanan) [inline deployment/setup_20.04.sh:88]: Could you please rename the variable ?

File: deployment/setup_20.04.sh [hunk 5]
```diff
@@ -76,8 +83,32 @@ cp /home/chatwoot/chatwoot/deployment/chatwoot.target /etc/systemd/system/chatwo
...
+    read -p 'What is your domain name server (chatwoot.domain.com for example) : ' domain_name
+    curl https://ssl-config.mozilla.org/ffdhe4096.txt >> /etc/ssl/dhparam
+    wget https://raw.githubusercontent.com/chatwoot/chatwoot/develop/deployment/nginx_chatwoot.conf
+    cp nginx_chatwoot.conf /etc/nginx/sites-available/nginx_chatwoot.conf
+    certbot certonly --nginx -d $domain_name
+    sed -i "s/chatwoot.domain.com/$domain_name/g" /etc/nginx/sites-available/nginx_chatwoot.conf
+    ln -s /etc/nginx/sites-available/nginx_chatwoot.conf /etc/nginx/sites-enabled/nginx_chatwoot.conf
+    systemctl restart nginx
+
+sudo -i -u chatwoot << EOF
```
Comment (vishnu-narayanan) [inline deployment/setup_20.04.sh:105]: Fix indentation.

File: deployment/setup_20.04.sh [hunk 6]
```diff
@@ -17,21 +18,27 @@ apt install -y \
...
-EOF
+addgroup chatwoot rvm
 
 pg_pass=$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 15 ; echo '')
 sudo -i -u postgres psql << EOF
 \set pass `echo $pg_pass`
 CREATE USER chatwoot CREATEDB;
 ALTER USER chatwoot PASSWORD :'pass';
 ALTER ROLE chatwoot SUPERUSER;
+UPDATE pg_database SET datistemplate = FALSE WHERE datname = 'template1';
```
Comment (vishnu-narayanan) [inline deployment/setup_20.04.sh:36]: Would it be best to have this change in another PR ?

**Review by vishnu-narayanan — COMMENTED** [review]
@iriseden Thank you for the PR. Could you please check the comments ?

File: deployment/setup_20.04.sh (hunk 6 above)
Comment (iriseden) [reply deployment/setup_20.04.sh:36]: Why ?

Comment (vishnu-narayanan) [reply deployment/setup_20.04.sh:36]: Unrelated change ?

File: deployment/setup_20.04.sh (hunk 4 above)
Comment (iriseden) [reply deployment/setup_20.04.sh]: Done

File: deployment/setup_20.04.sh (hunk 5 above)
Comment (iriseden) [reply deployment/setup_20.04.sh:102]: Done

File: deployment/setup_20.04.sh (hunk 6 above)
Comment (iriseden) [reply deployment/setup_20.04.sh:36]: No, otherwise there is an error on the type of encoding used

**Review by vishnu-narayanan — APPROVED** [review]

File: deployment/chatwoot_nginx.conf [hunk 7]
```diff
@@ -0,0 +1,47 @@
+server {
```
Comment (sojan-official) [inline deployment/chatwoot_nginx.conf:1]: we should probably rename this file to `nginx_chatwoot.conf` since that's the name used in the script.

Comment (iriseden) [reply deployment/chatwoot_nginx.conf]: Down, sorry for bad name ;)

**Comment by sojan-official** [comment]
@iriseden  thanks a lot for the changes ❤️

**Pull request merged**